WordPress/gutenberg
# In-between Inserter: Show inserter when it doesn't conflict with block toolbar

Fixes #53315

## What?

This PR relaxes the conditions under which in-between inserters appear. This will now allow in-between inserters to always appear in the following scenarios:

- Orientation is "horizontal"
- Top toolbar is enabled
- `__experimentalCaptureToolbars` is enabled

## Why?

To avoid conflicts with the block toolbar, we don't allow the in-between inserter to appear when hovering above the selected block. However, in the three contexts mentioned above, the inserter will not conflict with the block toolbar.

## Testing Instructions

Confirm the following scenarios:

Note: This PR doesn't fix the issue reported in #64222 where the in-between inserter doesn't always appear on wrapped blocks. I think we probably need to update [this condition](https://github.com/WordPress/gutenberg/blob/79b075293de8a763a5ea892406e53d92bfd2c63a/packages/block-editor/src/components/block-list/use-in-between-inserter.js#L146-L151) to fix #64222.

### Buttons (Horizontal):
- Insert a Buttons block and
- Select any Button block except the first one.
- Hover over the left side of the block.
- The inserter will appear because the orientation is horizontal.

### Buttons (Vertical):
- Insert a Buttons block.
- Change the direction to Vertical.
- Select any Button block except the first one.
- Hover above the button.
- The inserter will appear because `__experimentalCaptureToolbars` is enabled.

### Group (Top toolbar disabled): 
- Insert a Group block and then insert several blocks into it.
- Select any block except the first one.
- Hover above the button.
- The inserter will NOT be displayed because it conflicts with the block toolbar.

### Group (Top toolbar enabled): 
- Insert a Group block and then insert several blocks into it.
- Select any block except the first one.
- Hover above the button.
- The inserter will be displayed because it does NOT conflict with the block toolbar.

## Screenshots or screencast 

https://github.com/user-attachments/assets/7ec233a6-ccc1-4c5e-bc65-147cc613eea1
